dev/com.ibm.ws.microprofile.faulttolerance/src/com/ibm/ws/microprofile/faulttolerance/impl/ExecutionContextImpl.java

```diff
@@ -115,14 +135,110 @@ public long check() {
         return remaining;
     }
 
-    public void onRetry() {
-        this.retries++;
-        debugRelativeTime("onRetry: " + this.retries);
-        if (timeout != null) {
-            timeout.restart();
+    public void onRetry(Throwable t) {
+        try {
+            this.retries++;
+            debugRelativeTime("onRetry: " + this.retries);
+            metricRecorder.incrementRetriesCount();
+            if (timeout != null) {
+                timeout.restart();
+                attemptStartTime = System.nanoTime();
+            }
+            onAttemptComplete(t);
+        } catch (RuntimeException e) {
```

Comment:
check if failsafe catches throwable or exception

Reply:
It catches `Exception`